### PR TITLE
Move to latest analyzer/compiler toolset packages and fix violations …

### PR DESCRIPTION
**Pull request by mavasani**
<details><summary>Ask Mode template</summary>

### Customer scenario

Dogfood latest Roslyn analyzers on Roslyn.sln

### Bugs this fixes

N/A

### Workarounds, if any

N/A

### Risk

Low

### Performance impact

None

### Is this a regression from a previous update?

No

### Root cause analysis

N/A

### How was the bug found?

N/A

### Test documentation updated?

N/A

</details>

### CODE REVIEW

**Comment by mavasani** [comment]
@jinujoseph This PR is blocked on us validating and releasing the latest `2.6.0-beta2` analyzer packages on nuget.

File: build/Targets/Settings.props [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <VSIXExpInstallerPackageVersion>0.2.1-beta</VSIXExpInstallerPackageVersion>
     <RoslynDiagnosticsNugetPackageVersion>$(MicrosoftNetRoslynDiagnosticsVersion)</RoslynDiagnosticsNugetPackageVersion>
     <!-- Disable analyzers if NuGet deviated from the expected restore location (e.g. by changing case on a case-sensitive file system) -->
-    <UseRoslynAnalyzers Condition="!Exists('$(NuGetPackageRoot)\Microsoft.CodeQuality.Analyzers\$(RoslynDiagnosticsNugetPackageVersion)\analyzers\dotnet\cs\Analyzer.Utilities.dll')">False</UseRoslynAnalyzers>
```
Comment (jasonmalinowski) [inline build/Targets/Settings.props]: Can we just get rid of this hack now?

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs [hunk 2]
```diff
@@ -113,7 +113,9 @@ public AnalyzerExecutor(AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateS
                 // get compiler result builder map
                 var builderMap = analysisResult.ToResultBuilderMap(project, version, analyzerDriver.Compilation, analyzerDriver.Analyzers, cancellationToken);
 
+#pragma warning disable RS0012 // Do not call ToImmutableCollection on an ImmutableCollection value https://github.com/dotnet/roslyn-analyzers/issues/1430
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs]: Might be worth more explicitly stating "disabled due to analyzer bug" so I don't have to pull up a web browser to figure out what that bug is.

File: build/Targets/Settings.props (hunk 1 above)
Comment (mavasani) [reply build/Targets/Settings.props]: This is tracked by https://github.com/dotnet/roslyn/issues/21005 and https://github.com/dotnet/roslyn/issues/21005#issuecomment-317822553 explains why it is not trivial work.

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs (hunk 2 above)
Comment (mavasani) [reply src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs]: This is fixed in latest beta release, I will remove the suppression.

**Comment by mavasani** [comment]
Ping.. can I get one more review before requesting ask mode approval?

**Comment by mavasani** [comment]
Tagging @Pilchie for ask mode approval

**Comment by 333fred** [comment]
> ' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

This seems like an unchanged file except the newline?

---
Refers to: src/Features/VisualBasic/Portable/CodeFixes/CorrectNextControlVariable/CorrectNextControlVariableCodeFixProvider.vb:1 in 6697e04. [](commit_id = 6697e045253bce5b812c7b47cd0a846e7b0f79be, deletion_comment = False)

**Comment by mavasani** [comment]
@dotnet/roslyn-infrastructure It seems [windows_release_unit64_prtest](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_unit64_prtest/11316/) has passed but the status still shows `Started`. Is this known?

**Comment by Pilchie** [comment]
Just to confirm, this works with both 15.5.2 and 15.6 Preview 1, right?

**Comment by mavasani** [comment]
> Just to confirm, this works with both 15.5.2 and 15.6 Preview 1, right?

Yes, the analyzers and the compiler toolset should work on both of these.

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_unit64_prtest please

**Comment by Pilchie** [comment]
Approved - thanks!

**Pull request merged**